cli/lsp/language_server.rs

```diff
@@ -176,6 +179,64 @@ pub struct StateSnapshot {
   pub npm: Option<StateNpmSnapshot>,
 }
 
+struct CpuUsageWatchdog {
```

Comment:
Maybe this should be in a separate file? Maybe we should cargo feature flag this ?